### PR TITLE
Styling tweaks to Activities List

### LINKED ISSUE #2288: Activities List page style changes
change the styling of the Activities List page per the "before" and "after" images as well as the style change list.  This is a follow up to design task #2116

BEFORE

AFTER

### This task is done when...
 - [x] Changed out dividers from UI 4Pt Blue to 1pt Light Gray in the activity edit sections
 - [x] Remove divider between "4 Program Activities" header section and activity list, add in padding between the header and the list, about 32 pixels or so
 - [x] Black divider changed to 1pt light gray under section description and above "4 Program Activities" header section
 - [x] Changed "Remove" to Delete and converted text to red e31c3d
 - [x] Change Activity 1, Activity 2 etc. list text from Bold H4 to Semibold H4

### PR DESCRIPTION
**Pull request by radavis**
Resolves #2288 

### This pull request changes...

- a number of styling changes to the Activities List

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] The change has been documented
  - [ ] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [x] Design has approved the experience
- [x] Product has approved the experience

### CODE REVIEW

**Comment by radavis** [comment]
Screen Shot 2020-06-23 at 11 50 40 AM

**Comment by jeromeleecms** [comment]
@radavis When I highlight the "Edit" or "Delete" options with a screen reader, the screen reader only tells me "Edit" or "Delete" -- with the new styling, can you announce that which Activity would be edited or delete prior to selecting the link? The app doesn't currently do this, so if it's out of scope, we can spin this up as a separate issue too.

Based on this: https://www.section508.gov/content/guide-accessible-web-design-development#links_and_buttons

**Comment by radavis** [comment]
@jeromeleecms It seems like the best option to bring these ambiguous `Edit` and `Delete` buttons into compliance is to add `aria-label` attributes. I just made this change, so it should show up on the preview deploy, soon.

File: web/src/styles/scss/activity.scss [hunk 1]
```diff
@@ -13,3 +10,8 @@
     color: $color-error-dark;
   }
 }
+
+// Override font weight for heading within cms.gov Review Component
+.entry-details--review div h4.ds-u-font-weight--bold {
+  font-weight: 600 !important;
+}
```
Comment (pphillips-fearless) [inline web/src/styles/scss/activity.scss:17]: I like the idea of adding a comment when using !important.  I'll try to be sure to do this too

**Pull request merged**